pkg/configuration/configuration.go

```diff
@@ -0,0 +1,135 @@
+	IdentityProvider = "identity.provider"
+
+	// DefaultIdentityProvider the default value used for the identity provider (IdP) for newly created users
+	DefaultIdentityProvider = "rhd"
+)
+
+// Kubefed configuration constants
+const (
+	ClusterHealthCheckPeriod        = "cluster.healthcheck.period"
+	DefaultClusterHealthCheckPeriod = "10s"
+
+	ClusterHealthCheckTimeout        = "cluster.healthcheck.timeout"
+	DefaultClusterHealthCheckTimeout = "3s"
+
+	ClusterHealthCheckFailureThreshold        = "cluster.healthcheck.failure.threshold"
+	DefaultClusterHealthCheckFailureThreshold = 3
+
+	ClusterHealthCheckSuccessThreshold        = "cluster.healthcheck.success.threshold"
+	DefaultClusterHealthCheckSuccessThreshold = 1
+
+	ClusterAvailableDelay        = "cluster.available.delay"
+	DefaultClusterAvailableDelay = "20s"
+
+	ClusterUnavailableDelay        = "cluster.unavailable.delay"
+	DefaultClusterUnavailableDelay = "60s"
+)
+
+// Config encapsulates the Viper configuration registry which stores the
+// configuration data in-memory.
+type Config struct {
+	member *viper.Viper
+}
+
+// initConfig creates an initial, empty registry.
+func initConfig() *Config {
+	c := Config{
+		member: viper.New(),
+	}
+	c.member.SetEnvPrefix(MemberEnvPrefix)
+	c.member.AutomaticEnv()
+	c.member.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	c.setConfigDefaults()
+	return &c
+}
+
+func LoadConfig() *Config {
+	return initConfig()
+}
+
+func (c *Config) setConfigDefaults() {
+	c.member.SetTypeByDefaultValue(true)
+	c.member.SetDefault(IdentityProvider, DefaultIdentityProvider)
+	c.member.SetDefault(ClusterHealthCheckPeriod, DefaultClusterHealthCheckPeriod)
+	c.member.SetDefault(ClusterHealthCheckTimeout, DefaultClusterHealthCheckTimeout)
+	c.member.SetDefault(ClusterHealthCheckFailureThreshold, DefaultClusterHealthCheckFailureThreshold)
+	c.member.SetDefault(ClusterHealthCheckSuccessThreshold, DefaultClusterHealthCheckSuccessThreshold)
+	c.member.SetDefault(ClusterAvailableDelay, DefaultClusterAvailableDelay)
+	c.member.SetDefault(ClusterUnavailableDelay, DefaultClusterUnavailableDelay)
+}
+
+// GetAllMemberParameters returns the map with key-values pairs of parameters that have MEMBER_OPERATOR prefix
+func (c *Config) GetAllMemberParameters() map[string]string {
+	vars := map[string]string{}
+
+	for _, env := range os.Environ() {
+		keyValue := strings.SplitN(env, "=", 2)
+		if len(keyValue) < 2 {
+			continue
+		}
+		if strings.HasPrefix(keyValue[0], MemberEnvPrefix+"_") {
+			vars[keyValue[0]] = keyValue[1]
+		}
+	}
+	return vars
+}
+
+// GetIdP returns the configured Identity Provider (IdP) for the member operator
+// Openshift clusters can be configured with multiple IdPs. This config option allows admins to specify which IdP should be used by the toolchain operator.
+func (c *Config) GetIdP() string {
+	return c.member.GetString(IdentityProvider)
+}
+
+// GetClusterHealthCheckPeriod returns the configured member cluster health check period
```

Comment:
It's just a detail, so it's more FYI - the comment that you wrote may be understood as it is a period for member cluster health check in the sense that we want to check the member cluster, but it's vice versa. The period is used for checking a host cluster from the member cluster - in other words, the period is giving duration after which the Kubefed should check if the host cluster is ready or not. Similarly for the following methods.

Reply:
Done as part of ddbbefc0a4adaa34d70562e367889177f2cb83a3